src/app/[locale]/page.tsx

```diff
@@ -0,0 +1,118 @@
+'use client';
+
+import { Search } from '../../components/search/search';
+import { useEffect, useRef, useState } from 'react';
+import { Results } from '../../components/results/results';
+import type { DialogWindowHandle } from '../../components/dialog-window/dialog-window.types';
+import type { Character } from '../../components/card-list/card-list.types';
+import { Pagination } from '../../components/pagination/pagination';
+import type { PaginationProps } from '../../components/pagination/pagination.types';
+import { useSelector } from 'react-redux';
+import { SelectedItemsPanel } from '../../components/selected-items-panel/selected-items-panel';
+import type { RootState } from '../../store';
+import { useLazyGetCharactersQuery } from '../../services/api';
+import { useTranslations } from 'next-intl';
+import { MyContext } from '../../utils/CharacterContext';
+
+export default function Home() {
+  const heroesCount = 10;
+  const notFoundStatus = 404;
+  const serverErrorStatus = 500;
+  const [page, setPage] = useState(1);
+  const [fullData, setFullData] = useState<PaginationProps | null>(null);
+  const [items, setItems] = useState<Character[]>([]);
+  const [responseStatus, setResponseStatus] = useState<number | undefined>(
```

Comment:
guess it will be better to keep it `number | null` just to be consistence with `fullData`